ports/stm32/octospi.c

```diff
@@ -45,6 +45,24 @@
 
 void octospi_init(void) {
     // Configure OCTOSPI pins (allows 1, 2, 4 or 8 line configuration).
+    #if defined(STM32H7)
+    mp_hal_pin_config_alt_static_speed(MICROPY_HW_OSPIFLASH_CS, MP_HAL_PIN_MODE_ALT, MP_HAL_PIN_PULL_NONE, MP_HAL_PIN_SPEED_VERY_HIGH, STATIC_AF_OCTOSPIM_P1_NCS);
```

Comment:
Duplicating the code like this is no good.  Instead, I suggest making macros for the STATIC_AF part.  Eg:
```
#if defined(STM32H7)
#define STATIC_AF_OCTOSPI(signal) STATIC_AF_OCTOSPIM_P1_ ## signal
#else
#define STATIC_AF_OCTOSPI(signal) STATIC_AF_OCTOSPI1_ ## signal
#endif
```
then use it like `STATIC_AF_OCTOSPI(CLK)`.

Reply:
Thank you very much for your patience going through the codes and all the elegant suggestions to it! 

All of these changes has been done by now and I have made some changes to `system_stm32.c` to allow the clocksource selection. 

Tho, even I've tried to follow the instructions here [https://github.com/jimmo/git-and-micropython#my-pr-has-multiple-commits-2](https://github.com/jimmo/git-and-micropython#my-pr-has-multiple-commits-2) as much as I could, I think I've messed up the commit messages... If that was the case, please tell me what I need to do!